src/passes/inlining.cc

```diff
@@ -0,0 +1,109 @@
+#include "../internal.hh"
+#include "../utils.hh"
+
+namespace whilelang {
+    using namespace trieste;
+
+    PassDef inlining(
+        std::shared_ptr<CallGraph> &call_graph,
+        std::shared_ptr<ControlFlow> &cfg) {
```

Comment:
Since these are captured by the lambda further down, they should not be references. A reference to a shared_ptr does not increase its refcount, so if you keep the reference around (as in the lambda object below) it may refer to free'd memory.